e2e/nomostest/config-sync.go

```diff
@@ -1387,3 +1400,25 @@ func toMetav1Duration(t time.Duration) *metav1.Duration {
 		Duration: t,
 	}
 }
+
+// SetGitBranch updates the git branch
+func SetGitBranch(nt *NT, syncName, branch string) {
+	nt.T.Logf("Change git branch to %q", branch)
+	if nt.MultiRepo {
+		rs := fake.RootSyncObjectV1Beta1(syncName)
+		nt.MustMergePatch(rs, fmt.Sprintf(`{"spec": {"git": {"branch": "%s"}}}`, branch))
```

Comment:
`resetGitBranch` deleted the `reconciler-manager` pod. Do we need to do that here as well?

Reply:
Nah, the old func was only ever being called for mono-repo. So that code branch was never executed. For multi-repo updating the RSync is sufficient to have the reconciler-manager update the reconciler deployment, which replaces the pod automatically.

Reply:
Ack, thanks for explaining